src/vivarium/framework/results/interface.py

```diff
@@ -323,7 +321,7 @@ def register_adding_observation(
         additional_stratifications: list[str] = [],
         excluded_stratifications: list[str] = [],
         aggregator_sources: list[str] | None = None,
-        aggregator: Callable[[pd.DataFrame], float | pd.Series] = len,
+        aggregator: Callable[[pd.DataFrame], float | pd.Series[Any]] = len,
```

Comment:
Aren't the series' always number-likes?

Reply:
In this case they should always be number-like, since the results_updater is performing addition.